includes/Listeners/WonderCart.php

```diff
@@ -13,32 +13,156 @@ class WonderCart extends Listener {
 	 * @return void
 	 */
 	public function register_hooks() {
-		add_action('rest_after_insert_yith_campaign', array( $this, 'register_campaign' ), 10 );
+		add_action( 'rest_after_insert_yith_campaign', array( $this, 'register_campaign' ), 10 );
+		add_action( 'yith_sales_edit_campaign_event_modal_opened', array( $this, 'create_campaign_modal_open' ), 10, 2 );
+		add_action( 'yith_sales_edit_campaign_event_campaign_selected', array( $this, 'campaign_selected' ), 10, 2 );
+		add_action( 'yith_sales_edit_campaign_event_campaign_abandoned', array( $this, 'campaign_abandoned' ), 10, 2 );
+		add_action( 'woocommerce_checkout_order_processed', array( $this, 'checkout_campaigns_used' ), 10 );
 	}
 
 	/**
 	 * Campaign created
 	 *
-	 * @param string $post
+	 * @param string $post Campaign data
 	 *
 	 * @return string The post value
 	 */
-	public function register_campaign( $post){
-		$campaign   = yith_sales_get_campaign( $post->ID );
-		if ($campaign){
+	public function register_campaign( $post ) {
+		$campaign = yith_sales_get_campaign( $post->ID );
+		if ( $campaign ) {
 			$type = $campaign->get_type();
-			
+
 			$data = array(
-				"label_key"=> "type",
-				"type"=> $type,
+				'label_key' => 'type',
+				'type'      => $type,
```

Comment:
I see that we use `type` here and `campaign_type` everywhere else. Should we be using that here?

Reply:
In the newly added methods, campaign_type is used as the keyword.  This event was added earlier. I will confirm with Kasturi if we can update the keyword in this method too.

Reply:
@mamatharao05 Sounds good. Mention me here with what you find out, and I'll give this a final review.

Reply:
@wpscholar Kasturi suggested changing the keyword from 'campaign_type' to 'type' for all the events.